react/src/data/wrapper.ts

```diff
@@ -94,21 +100,29 @@ export class ModelsWrapper extends AbstractModelsWrapper<ModelState, unknown> {
     }
 
     hookModel = (modelName: string) => {
-        let selectFn = useSelector;
-        return selectFn(this.modelSelectors.selectModel(modelName));
+        let m = useSelector(this.modelSelectors.selectModel(modelName));
+        if(m === undefined || m === null) {
+            throw new Error("model could not be hooked because it was not found: " + modelName);
+        }
+        return m;
     }
 
     getFieldFromModel(fieldName: string, model: ModelState): unknown {
-        return model[fieldName];
+        let fv = model[fieldName];
+        if(fv === undefined || fv === null) {
+            throw new Error(`field could not be found in model state: ${fieldName}, ${JSON.stringify(model)}`)
```

Comment:
Is there a way to modularize this so that you don't have to write so much to describe the error?

Reply:
I am not sure. Writing error logging in JS is sort of difficult and we should probably use a library. Other languages make getting the relative context for a line of code easier than what I can find in JS. Besides that, I do not know what would be better to describe this without just writing it out. If I have some error message like "key not found in object" it doesn't really help anyone besides actually failing fast because it doesn't describe anything.

Reply:
In the angular app we send the stack to the server along with the error. This would definitely be useful. Is there an error handler in React similar [to AngularJS's](https://github.com/radiasoft/sirepo/blob/355d5e66fb769dfe51a4f3406409ba163ef7893f/sirepo/package_data/static/js/sirepo.js#L2972)?

Reply:
There doesn't seem to be an error handler within React for logging. There are ways we can handle errors during rendering but there isn't much I can find for logging using the React library. There are plenty of logging libraries though so we would just need to pick one.

Reply:
Please create an issue for this. We need to get the errors server side. Otherwise, they are lost.